### PR TITLE
Fix isMatchCustomizer return type fix

### PR DESCRIPTION
**Pull request by SKaistrenko**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://lodash.com/docs/4.17.15#isMatchWith
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/lodash/ts3.1/common/lang.d.ts [hunk 1]
```diff
@@ -838,7 +838,7 @@ declare module "../index" {
         isMap(): PrimitiveChain<boolean>;
     }
 
-    type isMatchCustomizer = (value: any, other: any, indexOrKey?: PropertyName) => boolean;
+    type isMatchCustomizer = (value: any, other: any, indexOrKey?: PropertyName) => boolean | undefined;
```
Comment (aj-r) [inline types/lodash/ts3.1/common/lang.d.ts:841]: It looks like this type is never used. Can we delete it instead?

Same for the non-3.1 version.

**Review by aj-r — CHANGES_REQUESTED** [review]

File: types/lodash/ts3.1/common/lang.d.ts (hunk 1 above)
Comment (SKaistrenko) [reply types/lodash/ts3.1/common/lang.d.ts:841]: It is used, for `isMatchWith`

Comment (SKaistrenko) [reply types/lodash/ts3.1/common/lang.d.ts:841]: `isMatchWith(object: object, source: object, customizer: isMatchWithCustomizer): boolean`

Comment (aj-r) [reply types/lodash/ts3.1/common/lang.d.ts:841]: That is `isMatchWithCustomizer`, not `isMatchCutsomizer`.

Comment (SKaistrenko) [reply types/lodash/ts3.1/common/lang.d.ts:841]: Oh, true. That's what I updated in my project when testing. Will update the `isMatchWithCustomizer` and remove this one

Comment (SKaistrenko) [reply types/lodash/ts3.1/common/lang.d.ts:841]: But seems like it's used in v3
    //_.isMatch
    interface isMatchCustomizer {
        (value: any, other: any, indexOrKey?: number|string): boolean;
    }

    interface LoDashStatic {
        /**
         * Performs a deep comparison between object and source to determine if object contains equivalent property
         * values. If customizer is provided it’s invoked to compare values. If customizer returns undefined
         * comparisons are handled by the method instead. The customizer is bound to thisArg and invoked with three
         * arguments: (value, other, index|key).
         * @param object The object to inspect.
         * @param source The object of property values to match.
         * @param customizer The function to customize value comparisons.
         * @param thisArg The this binding of customizer.
         * @return Returns true if object is a match, else false.
         */
        isMatch(object: Object, source: Object, customizer?: isMatchCustomizer, thisArg?: any): boolean;
    }

    interface LoDashImplicitObjectWrapper<T> {
        /**
         * @see _.isMatch
         */
        isMatch(source: Object, customizer?: isMatchCustomizer, thisArg?: any): boolean;
    }

Comment (SKaistrenko) [reply types/lodash/ts3.1/common/lang.d.ts:841]: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/lodash/v3/index.d.ts#L11287-L11312

Comment (aj-r) [reply types/lodash/ts3.1/common/lang.d.ts:841]: Oh I see, that explains where it came from.

In that case, let's:
-Remove `isMatchCustomizer` from this file
- Also remove it from `lodash/common/lang.d.ts`
- Fix it in v3 (and add/update a test if possible)

Comment (SKaistrenko) [reply types/lodash/ts3.1/common/lang.d.ts:841]: Why is there no `isMathcWith` in `/v3/index.d.ts`? https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/lodash/v3/index.d.ts#L11287-L11312

Comment (SKaistrenko) [reply types/lodash/ts3.1/common/lang.d.ts:841]: @aj-r

Comment (SKaistrenko) [reply types/lodash/ts3.1/common/lang.d.ts:841]: Oh, it's correct. There was no `isMatchWith` in v3.1. It was separated in v4

Comment (SKaistrenko) [reply types/lodash/ts3.1/common/lang.d.ts:841]: Is /ts3.1 - version of types for `lodash` 4.0+?

**Comment by aj-r** [comment]
Yes

**Review by aj-r — APPROVED** [review]

**Comment by SKaistrenko** [comment]
@aj-r Is there anything else needed from me to have this merged and pushed to NPM?

**Comment by aj-r** [comment]
Nope everything looks good

**Review by rbuckton — APPROVED** [review]

**Pull request merged**